protocol/messenger_status_updates.go

```diff
@@ -264,6 +268,16 @@ func (m *Messenger) HandleStatusUpdate(state *ReceivedMessageState, message *pro
 			return err
 		}
 		state.Response.AddStatusUpdate(statusUpdate)
+		if statusUpdate.StatusType == int(protobuf.StatusUpdate_AUTOMATIC) ||
+			statusUpdate.StatusType == int(protobuf.StatusUpdate_ALWAYS_ONLINE) ||
+			statusUpdate.StatusType == int(protobuf.StatusUpdate_INACTIVE) {
+			m.logger.Debug("reset data sync for peer", zap.String("public_key", statusUpdate.PublicKey), zap.Uint64("clock", statusUpdate.Clock))
+			m.mvdsStatusChangeEvent <- datasyncnode.PeerStatusChangeEvent{
```

Comment:
I think here you want to `select` so it doesn't block, just in case the buffer is full for whatever reason, as it would hang the processing of messages